src/routes/home.tsx

```diff
@@ -5,27 +5,55 @@ import Spinner from '@/components/spinner'
 import { compareEvents } from '@/utils/compare-events'
 import { useGeoLocation } from '@/utils/use-geo-location'
 import { useEvents } from '@/cache/events/use'
+import { useState } from 'preact/hooks'
 
 const now = new Date()
-
 const Home = () => {
   const events = useEvents()
   const location = useGeoLocation()
+  const [term = '', setTerm] = useState(0);
+
+  function searchingFor(term: string | number) {
+    return function (x: any) {
+      // because not all objects contain properties 'district' and 'fullDistrict' you have to check if particular object contains property 'distict' and if not tell method to not use it. Otherwise this error will occur (Uncaught (in promise) TypeError: Cannot read property 'toString' of undefined)
+      if (x.hasOwnProperty('district')) {
+        return (x.name.toString().toLowerCase().includes(term.toString().toLowerCase()) || !term) ||
+          (x.key.toString().toLowerCase().includes(term.toString().toLowerCase()) || !term) ||
+          (x.locationName.toString().toLowerCase().includes(term.toString().toLowerCase()) || !term) ||
+          (x.district.toString().toLowerCase().includes(term.toString().toLowerCase()) || !term) ||
+          (x.fullDistrict.toString().toLowerCase().includes(term.toString().toLowerCase()) || !term);
+      }
+
+      return (x.name.toString().toLowerCase().includes(term.toString().toLowerCase()) || !term) ||
+        (x.key.toString().toLowerCase().includes(term.toString().toLowerCase()) || !term) ||
+        (x.locationName.toString().toLowerCase().includes(term.toString().toLowerCase()) || !term);
```

Comment:
I think these nested conditionals could use some simplification. Here are some suggestions:

- You don't need to call `.toString()` on things that are already strings. If you are using VSCode, you can hover over a value (like `x.key`), and it will show you the type. If it is already `string`, calling `.toString()` is unnecessary.
- You are repeating `term.toString().toLowerCase()` many times. I'd suggest extracting this to a variable. Also we don't need to use `.toString` once we remove `number` from the type:
  ```js
  const lowercaseQuery = term.toLowerCase();
  ```
- You have `|| !term` on every single condition. I think it would be much simpler if we used a single early return at the start of the function instead. This would still handle the case of if the user has not typed anything, in which case we want to show all matches:
  ```js
  if (!term) return true;
  ```